ruleset.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<ruleset
+    name="Apex Security"
+    xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd"
+>
+   <description>Security Rules for Apex</description>
+
+   <rule
+        ref="category/apex/security.xml/ApexSharingViolations"
+        message="Apex classes should declare a sharing model if DML or SOQL is used"
+    >
+      <priority>3</priority>
+   </rule>
+   <rule
+        ref="category/apex/security.xml/ApexInsecureEndpoint"
+        message="Apex callouts should use encrypted communication channels"
+    >
+      <priority>3</priority>
+   </rule>
+   <rule ref="category/apex/errorprone.xml/ApexCSRF">
+      <priority>3</priority>
+   </rule>
+   <rule
+        ref="category/apex/security.xml/ApexOpenRedirect"
+        message="Apex classes should safely redirect to a known location"
+    >
+      <priority>3</priority>
+   </rule>
+   <rule
+        ref="category/apex/security.xml/ApexSOQLInjection"
+        message="Apex classes should escape variables merged in DML query"
+    >
+      <priority>3</priority>
+   </rule>
+   <rule
+        ref="category/apex/security.xml/ApexXSSFromURLParam"
+        message="Apex classes should escape Strings obtained from URL parameters"
+    >
+      <priority>3</priority>
+   </rule>
+   <rule
+        ref="category/apex/security.xml/ApexXSSFromEscapeFalse"
+        message="Apex classes should escape addError strings"
+    >
+      <priority>3</priority>
+   </rule>
+   <rule
+        ref="category/apex/security.xml/ApexBadCrypto"
+        message="Apex Crypto should use random IV/key"
+    >
+      <priority>3</priority>
+   </rule>
+  <!-- <rule
```

Comment:
Turning CRUD/FLS off as it is handled in the `CanTheUser.cls` for this project

Reply:
@msrivastav13 

I really don't want to do this. I think disabling a valid PMD rule, just because *we may* have called canTheUser first a disaster waiting to happen. Lets leave the rules as they are, and focus on what it might look like to make the PMD check aware of can the user?

Reply:
ok, that makes sense! For now, I am going to enable the rule as I agree can cause disaster!

To make PMD aware of the `canTheUser` would need some Xpath work and a new PMD rule that would be an interesting exercise to built on top of this change to use `ruleset.xml` instead of default. It will be a separate PR!